File: app/src/main/java/ardjomand/leonardo/nutrimeal/auth/login/LoginContract.java
package ardjomand.leonardo.nutrimeal.auth.login;

public interface LoginContract {

    interface Presenter {

        void setView(View view);

        void logIn(String email, String password);

        void getCurrentUser();
    }

    interface View {

        void showProgressBar(boolean visibility);

        void showLoginForm(boolean visibility);

        void goToMainScreen();

        void showError(String message);
    }
}

File: app/src/main/java/ardjomand/leonardo/nutrimeal/auth/login/LoginContract.kt
package ardjomand.leonardo.nutrimeal.auth.login

internal interface LoginContract {

    interface Presenter {

        fun setView(view: View?)

        fun logIn(email: String, password: String)

        fun getCurrentUser()
    }

    interface View {

        fun showProgressBar(visibility: Boolean)

        fun showLoginForm(visibility: Boolean)

        fun goToMainScreen()

        fun showError(message: String)
    }
}
